server/src/tests/api.test.js

```diff
@@ -0,0 +1,145 @@
+  });
+
+  describe("get", () => {
+    it("should responds with 200 and minutes body when id is valid and minutes are found", async () => {
+      const response = await api.get(`${baseUrl}/${minutesInDb.id}`);
+      expect(response.status).toBe(200);
+      expect(response.body).toEqual(expect.objectContaining(dummyMinutes));
+    });
+
+    it("should respond with 404 and error message when id doesn't exist in the database", async () => {
+      const response = await api.get(`${baseUrl}/65edcc9eb413f592f45351b1`);
+      expect(response.status).toBe(404);
+      expect(response.body.error).toEqual(
+        "Minutes not found with the given id",
+      );
+    });
+
+    it("should respond with 400 and validation error message when id isn't a valid mongodb id (hexadecimal value with 24 characters)", async () => {
+      const response = await api.get(`${baseUrl}/65edcc9e`);
+      expect(response.status).toBe(400);
+      expect(response.body.error).toEqual(
+        'Validation error: Invalid MongoDB id at "id"',
+      );
+    });
+  });
+
+  describe("post", () => {
+    it("should respond with 200 and the created minutes when given valid minutes object", async () => {
+      const response = await api.post(`${baseUrl}/`).send(dummyMinutes);
+      expect(response.status).toBe(200);
+    });
+
+    it("should respond with 400 and validation error message when given invalid minutes object in request body", async () => {
+      const response = await api
+        .post(`${baseUrl}/`)
+        .send({ ...dummyMinutes, name: undefined });
+      expect(response.status).toBe(400);
+      expect(response.body.error).toEqual(
+        'Validation error: Required at "name"',
+      );
+    });
+  });
+
+  describe("put", () => {
+    it("should respond with 200 and the updated minutes when given valid id and minutes body", async () => {
+      const updatedMinutes = { ...dummyMinutes, name: "updated minutes" };
+
+      const response = await api
+        .put(`${baseUrl}/${minutesInDb.id}`)
+        .send(updatedMinutes);
+
+      expect(response.status).toBe(200);
+      expect(response.body).toEqual(expect.objectContaining(updatedMinutes));
+    });
+
+    it("should respond with 400 and error message when given invalid id (hexadecimal value with 24 characters)", async () => {
+      const response = await api
+        .put(`${baseUrl}/65edcc9eb413f59`)
+        .send({ ...dummyMinutes, name: "updated minutes" });
+
+      expect(response.status).toBe(400);
+      expect(response.body.error).toEqual(
+        'Validation error: Invalid MongoDB id at "id"',
+      );
+    });
+
+    it("should respond with 400 and error message when given invalid body", async () => {
+      const response = await api
+        .put(`${baseUrl}/65edcc9eb413f59`)
+        .send({ ...dummyMinutes, name: undefined });
+
+      expect(response.status).toBe(400);
+      expect(response.body.error).toEqual(
+        'Validation error: Invalid MongoDB id at "id"',
```

Comment:
shouldnt the expected result be something like 'Validation error: Required at "name"'? So shouldnt the id that is put in in this be valid and just the body be invalid

Reply:
Good catch. The test is mostly just a copy paste of the test above it. I should check them one more time.